src/pages/Combat.tsx

```diff
@@ -121,43 +138,55 @@ export default function CombatPage({stats, onChange, onRoll, locked} : CombatPag
             </Initiative>
         </Grid>
         <Grid item xs={6}>
-
-                <Collection
-                    heft={calculateHeft(stats)}
-                    locked={locked}
-                    talents={stats.talents}
-                    values={characterWeapons}
-                    onChange={changeWeapons}
-                    onAction={performAction}
-                    component={Weapon} />
-
+            <Grid item xs={6} marginY={1}>Weapons</Grid>
+            <Collection
+                heft={calculateHeft(stats)}
+                locked={locked}
+                talents={stats.talents}
+                values={characterWeapons}
+                onChange={changeWeapons}
+                onAction={performAction}
+                component={Weapon}/>
+
+            {isWeaponAddVisible && !locked && (<Grid container direction="row" spacing={2} alignItems="center" marginY={1}>
+                <Grid item xs={8.7}><Dropdown
+                    id="add-weapon"
+                    label="Weapon Type"
+                    name="weapon"
+                    values={data}
+                    onChange={changeData}
+                    options={weapons.map(w => ({id: w.weapon, name: `${w.name} (${w.weapon})`}))}/></Grid>
+                <Grid item xs={2}><Btn fullWidth size="large" onClick={addWeapon}>Add</Btn></Grid>
+                <Grid item xs={1}><Btn fullWidth size="large" onClick={hideWeapon} color="error">X</Btn></Grid>
```

Comment:
Instead of using the letter `X`, you could see how the `×` looks like (https://www.amp-what.com/unicode/search/multipl). Maybe it's a little more pleasing to the eye. Alternatively, check out the https://mui.com/material-ui/icons/.